paypro-gateways-woocommerce/composer.json

```diff
@@ -0,0 +1,5 @@
+{
```

Comment:
This will only work with manual installation. If you download the plugin through the store it does not install `composer` as well. I think it would be preferred if we do not use composer here and use the manual loading.

The README of the repo explains how you can do that.

Reply:
I added the vendor files, so no composer commands are needed anymore.

Reply:
:+1: I will do a functional test and if that works we can merge the PR.

Reply:
@MichelOostinga 

Functional it works, but I was thinking if there are other options of including the API code. I know some other plugins that use `git submodules` to include the API code into this repository.

Since we allow including the API code by just requiring the `init.php` file we can also go for this route. This would remove a lot of composer files that we do not really need in the plugin.

What are your opnions on this?

Reply:
Git submodules could indeed be used, but they come with a few downsides as well, namely:
* If you're adding a submodule to an existing project, everyone that works on the project has to run commands like `git submodule update` and `git clone --recursive` to ensure they get everything—this includes any automated deployment or testing service that might be involved in the project.
* Collaborators won't automatically see updates to submodules—if you update a submodule, you may need to remind your colleagues to run `git submodule update` or they will likely see odd behavior.
* Managing dynamic, rapidly evolving or heavily co-dependent repositories with submodules can quickly become frustrating.

Also, the only composer dependency here is paypro API, so there aren't that many files inside the vendor folder.

I made a final little adjustment to the composer config, so that VCS won't break anything if new dependencies are added later on.

If you still prefer the `git submodule` option, then I think it would be best if you create a new repository ('paypro-dependencies', or something like that) to which I will create a pull request to add the dependency that way.